virttest/utils_libvirt/libvirt_network.py

```diff
@@ -81,3 +83,62 @@ def check_established(params):
 
     cmdRes = remote.run_remote_cmd(cmd, params)
     libvirt.check_result(cmdRes, expected_match=exp_msg)
+
+
+def modify_network_xml(net_dict, testnet_xml):
+    """
+    modify the network's xml
+
+    :param testnet_xml: the network xml object to be modified
+    :param net_dict: The dict restore need updated items like mac, bandwidth, forward
+    :return: the updated network xml
+    """
+    del_nat = net_dict.get('del_nat')
+    del_ip = net_dict.get('del_ip')
+    dns_txt = net_dict.get('dns_txt')
+    domain = net_dict.get('domain')
+    bridge = net_dict.get('bridge')
+    forward = net_dict.get('forward')
+    # get the params about forward interface
+    interface_dev = net_dict.get('interface_dev')
+    virtualport = net_dict.get('virtualport')
+    # bandwidth to be set if any
+    net_bandwidth_outbound = net_dict.get('net_bandwidth_outbound')
+    net_bandwidth_inbound = net_dict.get('net_bandwidth_inbound')
+    mac = net_dict.get("mac")
+
+    # delete the <bridge/> and <mac/> elements as they can be
+    # generated automatically if needed
+    testnet_xml.del_bridge()
+    testnet_xml.del_mac()
+    if del_nat:
+        testnet_xml.del_nat_attrs()
+    if del_ip:
+        testnet_xml.del_ip()
+    if dns_txt:
+        dns_dict = {"txt": ast.literal_eval(dns_txt)}
+        dns_obj = testnet_xml.new_dns(**dns_dict)
+        testnet_xml.dns = dns_obj
+    if mac:
+        testnet_xml.mac = mac
+    if domain:
+        testnet_xml.domain_name = domain
+    if bridge:
+        testnet_xml.del_bridge()
+        testnet_xml.bridge = {"name": bridge}
+    if forward:
+        testnet_xml.del_forward()
+
+        testnet_xml.forward = eval(forward)
+        logging.debug("current mode is %s" % testnet_xml.forward)
+    if interface_dev:
+        testnet_xml.forward_interface = [{'dev': interface_dev}]
+    if virtualport:
+        testnet_xml.virtualport_type = "openvswitch"
```

Comment:
you set a hardcode 'openvswitch' to virtualport_type if virtualport, is it correct?
Do we need to set virtualport as well?

Reply:
no need, libvirt will generate one automatically:
set: <virtualport type='openvswitch'/> 
and after the network start, it will generate a interfaceid(ovs-bridge-uuid) like:
<virtualport type='openvswitch'>
<parameters interfaceid='09b11c53-8b5c-4eeb-8f00-d84eaa0aaa4f'/>
</virtualport>

Reply:
virtualport = net_dict.get('virtualport')
if virtualport:
        testnet_xml.virtualport_type = virtualport

@Yingshun  What do you think about this? if there is no virtualport from the net_dict, we will not do this, if there is virtualport type, just assign it to 'type'.

Reply:
How about changing 'virtualport' to 'virtualport_type' directly? 
```
virtualport_type = net_dict.get('virtualport_type')
if virtualport_type:
    testnet_xml.virtualport_type = virtualport_type
```

Or making virtualport_type configurable? 
```
virtualport = net_dict.get('virtualport')
if virtualport:
    testnet_xml.virtualport_type = net_dict.get('virtualport_type', 'openvswitch')
```

Reply:
Updated, Thank you!